client/llb/source.go

```diff
@@ -199,13 +200,32 @@ type ImageInfo struct {
 
 func Git(remote, ref string, opts ...GitOption) State {
 	url := ""
+	isSSH := true
+	var sshHost string
 
 	for _, prefix := range []string{
-		"http://", "https://", "git://", "git@",
+		"http://", "https://",
```

Comment:
`git://` does not mean ssh. I think we should do a more specific regexp check for ssh `@:` . If no protocol then I think we should default to `git://`

Reply:
I had a read over https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols and never realised that there was a seperate `git://` protocol. I added a regex check to identify git over ssh. Let me know if that works.